rasa/graph_components/providers/domain_without_response_provider.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+from typing import Dict, Text, Any
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import KEY_RESPONSES, Domain
+
+
+class DomainWithoutResponseProvider(GraphComponent):
+    """Provides domain without information about responses."""
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> DomainWithoutResponseProvider:
+        """Creates component (see parent class for full docstring)."""
+        return cls()
+
+    def recreate_without_responses(self, domain: Domain) -> Domain:
```

Comment:
another nit: can we rename this to `provide` so that all providers have a similar signature? Makes it easier to generate the recipes 🙈

Reply:
```suggestion
    def provide(self, domain: Domain) -> Domain:
```